install.ps1

```diff
@@ -15,57 +13,28 @@ Please file an issue if you encounter any problems!
+    $TempDir = $Env:TEMP
+    $RepoUrl = "https://github.com/shuttle-hq/shuttle"
+    $CargoHome = if ($null -ne $Env:CARGO_HOME) { $Env:CARGO_HOME } else { "$HOME\.cargo" }
 
-	$CargoHome = if ($null -ne $Env:CARGO_HOME) { $Env:CARGO_HOME } else { "$HOME\.cargo" }
-	$TempDir = $Env:TEMP
-	$Arch = [Environment]::GetEnvironmentVariable("PROCESSOR_ARCHITECTURE", [EnvironmentVariableTarget]::Machine)
-	if (($Arch -eq "AMD64") -and (Get-Command -CommandType Application -ErrorAction SilentlyContinue tar.exe)) {
-	(Invoke-WebRequest "$RepoUrl/releases/latest" -Headers @{ "Accept" = "application/json" }).Content -match '"tag_name":"([^"]*)"' | Out-Null
-		$LatestRelease = $Matches.1
-		$BinaryUrl = "$RepoUrl/releases/download/$LatestRelease/cargo-shuttle-$LatestRelease-x86_64-pc-windows-msvc.tar.gz"
-		Invoke-WebRequest $BinaryUrl -OutFile "$TempDir\cargo-shuttle.tar.gz"
-		New-Item -ItemType Directory -Force "$TempDir\cargo-shuttle"
-		tar.exe -xzf "$TempDir\cargo-shuttle.tar.gz" -C "$TempDir\cargo-shuttle"
-		Move-Item -Force "$TempDir\cargo-shuttle\cargo-shuttle-x86_64-pc-windows-msvc-$LatestRelease\cargo-shuttle.exe" "$CargoHome\bin\cargo-shuttle.exe"
-		Remove-Item -Recurse -Force -ErrorAction SilentlyContinue "$TempDir\cargo-shuttle.tar.gz", "$TempDir\cargo-shuttle"
-		Write-Host "Installed cargo-shuttle, try running ``cargo shuttle --help``" -ForegroundColor Green
-		return
-	}
-	elseif ($Arch -ne "AMD64") {
-		Write-Host "Unsupported Architecture: Binaries are not currently built for $Arch, skipping manual binary download" -ForegroundColor Red
-	}
-	else {
-		Write-Host "Could not find tar.exe, skipping manual binary download (required to extract the release asset)" -ForegroundColor Red
-	}
+    if (Get-Command -CommandType Application -ErrorAction SilentlyContinue cargo-binstall.exe) {
+        Write-Host "Installing cargo-shuttle using cargo binstall"
+        cargo-binstall.exe cargo-shuttle --no-confirm
+        if ($?) {
+            Write-Host "Installed cargo-shuttle, try running ``cargo shuttle --help``" -ForegroundColor Green
+            return
+        }
+        else {
+            Write-Host "Could not install from release using cargo binstall, trying manual binary download" -ForegroundColor Red
+        }
+    }
+    else {
+        Write-Host "Cargo binstall not found, trying manual binary download" -ForegroundColor Red
+    }
 
-	if (Get-Command -CommandType Application -ErrorAction SilentlyContinue cargo.exe) {
-		Write-Host "Installing cargo-shuttle using cargo install (from source)"
-		cargo.exe install cargo-shuttle --locked
-		if ($?) {
-			Write-Host "Installed cargo-shuttle, try running ``cargo shuttle --help``" -ForegroundColor Green
-			return
-		}
-		else {
-			Write-Host "Could not install cargo-shuttle using cargo" -ForegroundColor Red
-			return
-		}
-	}
-	else {
-		if ($Arch -in "AMD64", "x86") {
+    if (-not (Get-Command -CommandType Application -ErrorAction SilentlyContinue cargo.exe)) {
```

Comment:
Can we move the cargo check + rust installation to before the binstall step, like we do in the `sh` script?

Reply:
[d696ec6](https://github.com/shuttle-hq/shuttle/pull/1636/commits/d696ec66189aea6810848b91fb203d4806e288aa)